src/tickit/core/simulation.py

```diff
@@ -0,0 +1,135 @@
+                yield (
+                    asyncio.create_task(
+                        component.run_forever(*get_interface(self._backend))
+                    )
+                )
+
+
+class TickitSimulationBuilder:
+    """A Ticket simulation builder.
+
+    A utility class that uses a config file to create a scheduler and the components
+    for a simulation.
+    """
+
+    _backend: str
+    _config_path: str
+
+    _include_schedulers: bool
+    _include_components: bool
+    _components_to_run: Set[ComponentID]
+
+    def __init__(
+        self,
+        config_path: str,
+        backend: str = "internal",
+        include_schedulers: bool = True,
+        include_components: bool = True,
+        components_to_run: Optional[Set[ComponentID]] = None,
+    ) -> None:
+        """Instantiate the builder class variables.
+
+        Args:
+            config_path (str): The path to the configuration file.
+            backend (str): The message broker to be used.
+            include_schedulers (bool): A flag to determine if the master scheduler is
+                included in the simulation. Defaults to True.
+            include_components (bool): A flag to determine if components are included
+                in the simulation. Defaults to True.
+            components_to_run (Optional[Set[ComponentID]): A subset of components in
+                the system to be run. Defaults to None, in which case all components
+                will be run.
+
+        """
+        self._backend = backend
+        self._config_path = config_path
+
+        self._include_schedulers = include_schedulers
+        self._include_components = include_components
+        self._components_to_run = components_to_run or set()
+
+    def build(self) -> TickitSimulation:
+        """Builds the master scheduler and components for a simulation.
+
+        A config file is read, retrieving the relevant components. The wiring of
+        these components is generated and used by the master scheduler for event
+        routing.
+
+        Returns:
+            TickitSimulation: A simulation object containing a given set of a scheduler
+                and components which can be run.
+
+        """
+        configs = read_configs(self._config_path)
+        inverse_wiring = InverseWiring.from_component_configs(configs)
+        scheduler = MasterScheduler(inverse_wiring, *get_interface(self._backend))
+        components = {config.name: config() for config in configs}
+
+        components = {
+            config.name: config()
+            for config in configs
+            if config.name
+            in (
+                self._components_to_run
+                if self._components_to_run
+                else (config.name for config in configs)
+            )
+        }
+
+        return TickitSimulation(
+            self._backend,
+            scheduler if self._include_schedulers else None,
+            components if self._include_components else None,
```

Comment:
Could we avoid creating the things we don't need or are there side effects that we're relying on?

Reply:
I could put some if statements around the scheduler and component creation and leave these lines as is. So we still get them when we want them and `None` if we don't? Or would you propose something else?

Reply:
If it works but it might throw up type errors if you leave these lines as they are. Could use something like 
```python
scheduler = None
if include_schedulers:
    scheduler = ....
# same for components
return TickitSimulation(backend, schedulers, components)
```